src/block/parser.rs

```diff
@@ -176,12 +180,16 @@ impl BlockParser {
                     }
                 }
 
-                let successive_idx = length_start_indices[curr_start_idx + 1];
+                let successive_idx = if curr_start_idx + 1 < length_start_indices.len() {
+                    Some(length_start_indices[curr_start_idx + 1])
+                } else {
+                    None
+                };
 
                 // curr_path represents a canonical block
                 info!(
-                    "Found canonical tip {} in {:?}",
-                    curr_path.file_name().unwrap().to_str().unwrap(),
+                    "Found canonical tip with hash {} in {:?}",
+                    get_state_hash(curr_path.file_name().unwrap()).unwrap(),
```

Comment:
Can we return a `Result` from this function and use `?`'s instead of repeatedly calling `unwrap` everywhere?

Reply:
Added helpers